### PR TITLE
Add CI framework for Windows 64

### PR DESCRIPTION
**Pull request by samuel40791765**
*Issue #, if available:*
`CryptoAlg-1104`

*Description of changes:*
Though official support for Windows (64-bit) is blocked on https://github.com/corretto/amazon-corretto-crypto-provider/issues/48, we can still CI support for it. 
Once the Windows CI framework is added to ACCP, we only have to reconfigure the `tests/ci/run_windows_tests.bat` test script for ACCP windows testing, then enable the CI support.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/ci/cdk/util/metadata.py [hunk 1]
```diff
@@ -11,8 +11,15 @@
...
+WINDOWS_X86_ECR_REPO = EnvUtil.get("ECR_WINDOWS_X86_REPO_NAME", "accp-docker-images-windows-x86")
 
 # Used when AWS CodeBuild needs to create web_hooks.
 GITHUB_REPO_OWNER = EnvUtil.get("GITHUB_REPO_OWNER", "corretto")
 GITHUB_REPO_NAME = EnvUtil.get("GITHUB_REPO_NAME", "amazon-corretto-crypto-provider")
 GITHUB_SOURCE_VERSION = EnvUtil.get("GITHUB_SOURCE_VERSION", "develop")
+
+# Used when AWS CDK defines resources for Windows docker image build.
+S3_BUCKET_NAME = EnvUtil.get("S3_FOR_WIN_DOCKER_IMG_BUILD", "accp-windows-docker-image-build")
+WIN_EC2_TAG_KEY = EnvUtil.get("WIN_EC2_TAG_KEY", "accp")
```
Comment (WillChilds-Klein) [inline tests/ci/cdk/util/metadata.py:23]: `accp` as tag key? what's the motivation behind that tag name?

File: tests/ci/run_windows_tests.bat [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+set BUILD_DIR=%SRC_ROOT%\test_build_dir
+
+@rem This script was taken from AWS-LC and can be adjusted when ACCP adds support for Windows.
+@rem %1 contains the path to the setup batch file for the version of of visual studio that was passed into AWS-LC's build spec file.
+@rem x64 comes from the architecture options https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line
+set MSVC_PATH=%1
+call %MSVC_PATH% x64 || goto error
+SET
+
+call :build_and_test test "" || goto error
```
Comment (WillChilds-Klein) [inline tests/ci/run_windows_tests.bat:12]: where is `:build_and_test` defined? is that line 21?

**Review by samuel40791765 — COMMENTED** [review]
I've verified the Windows docker image building portion works, along with the CI set up. I've attached a file which shows the docker image building output, along with a screenshot of the images living in the appropriate ECR. Along with that, the CI changes are tested in this local fork PR: https://github.com/samuel40791765/amazon-corretto-crypto-provider/pull/15

Running a test command  with the script in the docker image outputs a gradle error during linking MSVC with AWS-LC. This would sound like one of the tasks to fix for ACCP Windows support, so I've stopped here.
```
C:\codebuild\tmp\output\src862177151\src\github.com\samuel40791765\amazon-corretto-crypto-provider>gradlew.bat release "-DTEST_JAVA_MAJOR_VERSION=17"   \|\| goto error
--
135 | Downloading https://services.gradle.org/distributions/gradle-7.2-bin.zip
136 | ..........10%...........20%...........30%...........40%...........50%...........60%...........70%...........80%...........90%...........100%
137 |  
138 | Welcome to Gradle 7.2!
139 |  
140 | Here are the highlights of this release:
141 | - Toolchain support for Scala
142 | - More cache hits when Java source files have platform-specific line endings
143 | - More resilient remote HTTP build cache behavior
144 |  
145 | For more details see https://docs.gradle.org/7.2/release-notes.html
146 |  
147 | Starting a Gradle Daemon (subsequent builds will be faster)
148 | > Task :assemble UP-TO-DATE
149 |  
150 | > Task :buildAwsLc
151 | -- Building for: Visual Studio 15 2017
152 | CMake Error at CMakeLists.txt:4 (project):
153 | Failed to run MSBuild command:
154 | -- Configuring incomplete, errors occurred!
155 |  
156 | See also "C:/codebuild/tmp/output/src862177151/src/github.com/samuel40791765/amazon-corretto-crypto-provider/build/awslc/build/CMakeFiles/CMakeOutput.log".
157 | C:/Program Files (x86)/Microsoft Visual Studio/2017/BuildTools/MSBuild/15.0/Bin/MSBuild.exe
158 |  
159 | to get the value of VCTargetsPath:
160 |  
161 | Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
162 | Copyright (C) Microsoft Corporation. All rights reserved.
163 |  
164 | Build started 7/9/2022 4:10:51 AM.
165 | Project "C:\codebuild\tmp\output\src862177151\src\github.com\samuel40791765\amazon-corretto-crypto-provider\build\awslc\build\CMakeFiles\3.15.4\VCTargetsPath.vcxproj" on node 1 (default targets).
166 | C:\codebuild\tmp\output\src862177151\src\github.com\samuel40791765\amazon-corretto-crypto-provider\build\awslc\build\CMakeFiles\3.15.4\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
167 | Done Building Project "C:\codebuild\tmp\output\src862177151\src\github.com\samuel40791765\amazon-corretto-crypto-provider\build\awslc\build\CMakeFiles\3.15.4\VCTargetsPath.vcxproj" (default targets) -- FAILED.
168 |  
169 | Build FAILED.
170 |  
171 | "C:\codebuild\tmp\output\src862177151\src\github.com\samuel40791765\amazon-corretto-crypto-provider\build\awslc\build\CMakeFiles\3.15.4\VCTargetsPath.vcxproj" (default target) (1) ->
172 | C:\codebuild\tmp\output\src862177151\src\github.com\samuel40791765\amazon-corretto-crypto-provider\build\awslc\build\CMakeFiles\3.15.4\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
173 |  
174 | 0 Warning(s)
175 | 1 Error(s)
176 |  
177 | Time Elapsed 00:00:00.40
178 |  
179 |  
180 | Exit code: 1
181 |  
182 |  
183 |  
184 |  
185 | > Task :buildAwsLc FAILED
186 |  
187 | FAILURE: Build failed with an exception.
188 |  
189 | * Where:
190 | Build file 'C:\codebuild\tmp\output\src862177151\src\github.com\samuel40791765\amazon-corretto-crypto-provider\build.gradle' line: 75
191 |  
192 | * What went wrong:
193 | Execution failed for task ':buildAwsLc'.
194 | > Process 'command 'cmake'' finished with non-zero exit value 1
```

Docker image building: [nohup.txt](https://github.com/corretto/amazon-corretto-crypto-provider/files/9087005/nohup.txt)
ECR: Screen Shot 2022-07-08 at 5 35 04 PM

File: tests/ci/run_windows_tests.bat (hunk 2 above)
Comment (WillChilds-Klein) [reply tests/ci/run_windows_tests.bat]: gotcha, thanks.

**Review by WillChilds-Klein — APPROVED** [review]
looks great, thanks sam!

**Pull request merged**